gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/SpecCatalog.java

```diff
@@ -55,24 +68,67 @@
     public static final String SPEC_ADDED_OPERATION_TYPE = "SpecAdded";
     public static final String SPEC_DELETED_OPERATION_TYPE = "SpecDeleted";
     public static final String SPEC_UPDATED_OPERATION_TYPE = "SpecUpdated";
+    public static final String TIME_FOR_SPEC_CATALOG_GET = "timeForSpecCatalogGet";
+    public static final String HISTOGRAM_FOR_SPEC_ADD = "histogramForSpecAdd";
+    public static final String HISTOGRAM_FOR_SPEC_UPDATE = "histogramForSpecUpdate";
+    public static final String HISTOGRAM_FOR_SPEC_DELETE = "histogramForSpecDelete";
 
-    @Getter
-    private final ContextAwareGauge<Integer> numActiveSpecs;
+    @Getter private final ContextAwareGauge<Integer> numActiveSpecs;
     @Getter private final ContextAwareCounter numAddedSpecs;
     @Getter private final ContextAwareCounter numDeletedSpecs;
     @Getter private final ContextAwareCounter numUpdatedSpecs;
-
-    public StandardMetrics(final SpecCatalog parent) {
-      this.numAddedSpecs = parent.getMetricContext().contextAwareCounter(NUM_ADDED_SPECS);
-      this.numDeletedSpecs = parent.getMetricContext().contextAwareCounter(NUM_DELETED_SPECS);
-      this.numUpdatedSpecs = parent.getMetricContext().contextAwareCounter(NUM_UPDATED_SPECS);
-      this.numActiveSpecs = parent.getMetricContext().newContextAwareGauge(NUM_ACTIVE_SPECS_NAME,
+    @Getter private final ContextAwareTimer timeForSpecCatalogGet;
+    @Getter private final ContextAwareHistogram histogramForSpecAdd;
+    @Getter private final ContextAwareHistogram histogramForSpecUpdate;
+    @Getter private final ContextAwareHistogram histogramForSpecDelete;
+
+    public StandardMetrics(final SpecCatalog specCatalog) {
+      this.timeForSpecCatalogGet = specCatalog.getMetricContext().contextAwareTimerWithSlidingTimeWindow(TIME_FOR_SPEC_CATALOG_GET, 1, TimeUnit.MINUTES);
```

Comment:
Same here. Using a local variable for `specCatalog.getMetricContext()` makes code cleaner